src/ucs/type/status.h

```diff
@@ -42,9 +49,23 @@ typedef enum {
     UCS_ERR_TIMED_OUT              = -20,
     UCS_ERR_EXCEEDS_LIMIT          = -21,
     UCS_ERR_UNSUPPORTED            = -22,
+
+    UCS_ERR_FIRST_LINK_FAILURE     = -40,
+    UCS_ERR_LAST_LINK_FAILURE      = -59,
+    UCS_ERR_FIRST_ENDPOINT_FAILURE = -60,
+    UCS_ERR_LAST_ENDPOINT_FAILURE  = -79,
+    UCS_ERR_ENDPOINT_TIMEOUT       = -80,
+
     UCS_ERR_LAST                   = -100
 } UCS_S_PACKED ucs_status_t ;
 
+#define USC_IS_LINK_ERROR(code) \
```

Comment:
1. please use leading _ for variable names
2. please surround _code with () when using inside the macro